l1-contracts/contracts/bridge/AssetRouterBase.sol

```diff
@@ -0,0 +1,268 @@
+        require(address(_nativeTokenVault) != address(0), "AR: native token vault 0");
+        nativeTokenVault = _nativeTokenVault;
+    }
+
+    /// @notice Sets the asset handler address for a given asset ID.
+    /// @dev No access control on the caller, as msg.sender is encoded in the assetId.
+    /// @param _assetData In most cases this parameter is bytes32 encoded token address. However, it can include extra information used by custom asset handlers.
+    /// @param _assetHandlerAddress The address of the asset handler, which will hold the token of interest.
+    function setAssetHandlerAddress(bytes32 _assetData, address _assetHandlerAddress) external virtual {
+        address sender = msg.sender == address(nativeTokenVault) ? L2_NATIVE_TOKEN_VAULT_ADDRESS : msg.sender;
+        bytes32 assetId = keccak256(abi.encode(uint256(block.chainid), sender, _assetData));
+        assetHandlerAddress[assetId] = _assetHandlerAddress;
+        emit AssetHandlerRegistered(assetId, _assetHandlerAddress, _assetData, sender);
+    }
+
+    /// @notice Allows bridgehub to acquire mintValue for L1->L2 transactions.
+    /// @dev If the corresponding L2 transaction fails, refunds are issued to a refund recipient on L2.
+    /// @param _chainId The chain ID of the ZK chain to which deposit.
+    /// @param _assetId The deposited asset ID.
+    /// @param _prevMsgSender The `msg.sender` address from the external call that initiated current one.
+    /// @param _amount The total amount of tokens to be bridged.
+    function bridgehubDepositBaseToken(
+        uint256 _chainId,
+        bytes32 _assetId,
+        address _prevMsgSender,
+        uint256 _amount
+    ) external payable override {
+        address assetHandler = assetHandlerAddress[_assetId];
+        _transferAllowanceToNTV(_assetId, _amount, _prevMsgSender);
+        // slither-disable-next-line unused-return
+        IAssetHandler(assetHandler).bridgeBurn{value: msg.value}({
+            _chainId: _chainId,
+            _mintValue: _amount,
+            _assetId: _assetId,
+            _prevMsgSender: _prevMsgSender,
+            _data: abi.encode(_amount, address(0))
+        });
+
+        // Note that we don't save the deposited amount, as this is for the base token, which gets sent to the refundRecipient if the tx fails
+        emit BridgehubDepositBaseTokenInitiated(_chainId, _prevMsgSender, _assetId, _amount);
+    }
+
+    /// @notice Initiates a transfer transaction within Bridgehub, used by `requestL2TransactionTwoBridges`.
+    /// @param _chainId The chain ID of the ZK chain to which deposit.
+    /// @param _prevMsgSender The `msg.sender` address from the external call that initiated current one.
+    /// @param _value The `msg.value` on the target chain tx.
+    /// @param _data The calldata for the second bridge deposit.
+    /// @return request The data used by the bridgehub to create L2 transaction request to specific ZK chain.
+    function bridgehubTransfer(
```

Comment:
I'm not sure we can change this name, does bridgehubDeposit not work here?

Reply:
There might be already bridges on L1 that are called from the bridgehub via bhDeposit